wizard

mantis-toboggan-md opened the pull request:
Just the `Wizard` component from #830

vincent99 commented on components/Wizard.vue:
```diff
@@ -0,0 +1,308 @@
+    }
+  },
+
+  asyncData(ctx) {
+    const { route:{ query } } = ctx;
+
+    return { queryStep: query.step };
+  },
```
Components don't have an `asyncData`, so this can't work?  But you can do this + what `data` does in `fetch`

vincent99 commented on components/Wizard.vue:
```diff
@@ -0,0 +1,308 @@
+  created() {
+    const { $route: { query = {} } } = this;
+
+    // if any step has active===true, load that step. Otherwise, load step defined in query param. Otherwise, load first step.
```
Comment no longer true

vincent99 commented on components/Wizard.vue:
```diff
@@ -0,0 +1,308 @@
+      }
+
+      if (queryStep !== number) {
+        this.$router.push({ ...this.$route, query: { ...this.$route.query, step: number } });
```
Use `STEP` from `config/query-params` everywhere instead of literal string so usage is easier to find.

 [applyQuery](https://github.com/rancher/dashboard/blob/master/plugins/extend-router.js#L4) is a shorter equivalent of this, and supports having lack of a query param for the "default" value of step=1:

```javascript
this.$router.applyQuery({[STEP]: number}, {[STEP]: 1})
```

mantis-toboggan-md commented:
In addition to addressing the above feedback, I added a title/image to the wizard banner, and a `ButtonSelect` component to be used when the 'next' button has alternative actions.

![Screen Shot 2020-07-16 at 3 45 42 PM](https://user-images.githubusercontent.com/42977925/87730499-d1ec0700-c77c-11ea-9473-d7119636e71d.png)
